### PR TITLE
Scoreboard locking

### PR DESCRIPTION
**Pull request by crast**

### CODE REVIEW

File: src/app/Api/Context.php [hunk 1]
```diff
@@ -76,21 +76,29 @@ public function setRegion(Models\Region $region)
...
     }
 
-    public function getRawSetting($name, $center = null)
+    public function getRawSetting($name, $center = null, $quarter = null)
     {
         if ($center == null) {
             $center = $this->getCenter();
         }
-        $setting = Models\Setting::get($name, $center);
+        if ($quarter == null) {
```
Comment (pdarg) [inline src/app/Api/Context.php]: The quarter arg in `Models\Setting::get()` defaults to null, so you could simplify this to just ```$setting = Models\Setting::get($name, $center, $quarter);```

Comment (crast) [inline src/app/Api/Context.php]: I thought you can't pass null explicitly to a typed parameter value, only default it to null

Comment (pdarg) [inline src/app/Api/Context.php]: I'm pretty sure you can, but you can only pass null to a typed parameter if it has a default value of null

File: src/app/Domain/ScoreboardLockQuarter.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        }
+
+        return ['weeks' => $weeksArray];
+    }
+
+    public static function fromArray($data)
+    {
+        $v = new static();
+        foreach ($data['weeks'] as $weekRaw) {
+            $d = Carbon::createFromFormat('Y-m-d', $weekRaw['week']);
```
Comment (pdarg) [inline src/app/Domain/ScoreboardLockQuarter.php]: You could simplify this to `Carbon::parse($weekRaw['week']);`

File: src/app/Setting.php [hunk 3]
```diff
@@ -38,10 +38,10 @@ public static function get($name, Center $center = null, Quarter $quarter = null
         }
 
         if ($center) {
-            $setting = Setting::byCenter($center)
+            $setting = self::byCenter($center)
```
Comment (pdarg) [inline src/app/Setting.php]: you use static on line 81 for something similar this. Do you want `self` or `static` here?

File: src/app/Setting.php [hunk 4]
```diff
@@ -52,10 +52,10 @@ public static function get($name, Center $center = null, Quarter $quarter = null
         }
 
         if (!$setting) {
-            $setting = Setting::whereNull('center_id')
+            $setting = self::whereNull('center_id')
```
Comment (pdarg) [inline src/app/Setting.php]: you use static on line 81 for something similar this. Do you want `self` or `static` here?

**Comment by pdarg** [comment]
Added a few comments, mostly nit-picky. Looks great

File: src/app/Setting.php (hunk 4 above)
Comment (crast) [inline src/app/Setting.php]: that's just a weird thing about phpfmt... it replaces references to the same class with self

Comment (crast) [inline src/app/Setting.php]: I don't want either, but it's not going to let me leave the class reference in

**Pull request merged**